Src/AutoMoq/MockType.cs

```diff
@@ -68,11 +68,14 @@ internal static Type GetMockedType(this Type type)
             where TMock : class
         {
             return setup.Returns(() =>
-                {
-                    var result = (TResult) context.Resolve(typeof (TResult));
-                    setup.Returns(result);
-                    return result;
-                });
+            {
+                var specimen = context.Resolve(typeof (TResult));
+                TResult result = specimen is TResult ? (TResult) specimen : default(TResult);

```

Comment:
Converting `OmitSpecimen` to `default(T)` shouldn't be the responsibility of an Auto-Mocking Container.

`OmitSpecimen` exists for the purpose of enabling recursion handlers and auto-property commands to skip the value. Prematurely converting the value takes away that ability.

Thus, I'm inclined to think that this conversion is undesirable.

What is the high-level scenario where this conversion is necessary?